pkg/cmd/workflow/disable/disable.go

```diff
@@ -0,0 +1,98 @@
+package disable
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmd/workflow/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type DisableOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	Selector string
+	Prompt   bool
+}
+
+func NewCmdDisable(f *cmdutil.Factory, runF func(*DisableOptions) error) *cobra.Command {
+	opts := &DisableOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+	}
+
+	cmd := &cobra.Command{
+		Use:    "disable [<workflow ID> | <workflow name>]",
+		Short:  "Disable a workflow",
+		Args:   cobra.MaximumNArgs(1),
+		Hidden: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			if len(args) > 0 {
+				opts.Selector = args[0]
+			} else if !opts.IO.CanPrompt() {
+				return &cmdutil.FlagError{Err: errors.New("workflow ID or name required when not running interactively")}
+			} else {
+				opts.Prompt = true
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return runDisable(opts)
+		},
+	}
+
+	return cmd
+}
+
+func runDisable(opts *DisableOptions) error {
+	c, err := opts.HttpClient()
+	if err != nil {
+		return fmt.Errorf("could not build http client: %w", err)
+	}
+	client := api.NewClientFromHTTP(c)
+
+	repo, err := opts.BaseRepo()
+	if err != nil {
+		return fmt.Errorf("could not determine base repo: %w", err)
+	}
+
+	var workflow *shared.Workflow
+
+	if opts.Prompt {
+		workflow, err = shared.PromptWorkflows(client, repo, shared.Active)
+		if err != nil {
+			return err
+		}
+	}
+
+	if workflow == nil {
```

Comment:
I commented the same thing on the workflow view PR. I think this condition is better expressed as `else` or `!opts.Prompt`.